examples/code_generation/config.py

```diff
@@ -0,0 +1,195 @@
+"""Various configuration options for the code generation task.
+
+This file is intended to be modified. You can go in and change any
+of the variables to run different experiments.
+"""
+
+from __future__ import annotations
+
+import transformers
+
+from zeno_build.evaluation.text_features.exact_match import avg_exact_match, exact_match
+from zeno_build.evaluation.text_features.length import (
+    input_length,
+    label_length,
+    output_length,
+)
+from zeno_build.evaluation.text_metrics.critique import (
+    avg_bert_score,
+    avg_chrf,
+    avg_length_ratio,
+    bert_score,
+    chrf,
+    length_ratio,
+)
+from zeno_build.evaluation.text_metrics.huggingface import (
+    execution_accuracy,
+    avg_execution_accuracy,
+)
+from zeno_build.experiments import search_space
+from zeno_build.models.dataset_config import DatasetConfig
+from zeno_build.models.lm_config import LMConfig
+
+
+# Define the space of hyperparameters to search over.
+space = search_space.CombinatorialSearchSpace(
+    {
+        "dataset_preset": search_space.Constant("odex"),
+        "model_preset": search_space.Categorical(
+            [
+                # "gpt-3.5-turbo",
+                # "cohere-command-xlarge",
+                # "gpt2",
+                # "gpt2-xl",
+                # "llama-7b",
+                # "alpaca-7b",
+                "codegen-350M-mono"
+                # "codegen-2B-mono"
+            ]
+        ),
+        "prompt_preset": search_space.Categorical(
+            ["standard"]
+        ),
+        "temperature": search_space.Discrete([0.2]),
+        "max_tokens": search_space.Constant(512),
+        "top_p": search_space.Constant(0.95),
+    }
+)
+
+# The number of trials to run
+num_trials = 1
+
+# The details of each dataset
+dataset_configs = {
+    "odex": DatasetConfig(
+        dataset="neulab/odex",
+        split="test",
+        data_column=["intent", "prompt"],
+        label_column=["test_start", "test", "entry_point"],
```

Comment:
Here `data_column` and `label_column` should be expressed as tuples, not lists, since the number of elements is fixed and won't change.

You will also probably need to change the typing information for `DatasetConfig` to accept tuples:
https://github.com/zeno-ml/zeno-build/blob/c59fb35baed113de66441b9f0be476f475d92b39/zeno_build/models/dataset_config.py#L25-L27